File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        maps={}
        n=len(nums)
        for i in nums:
            if(i in maps):
                maps[i]+=1
            else:
                maps[i]=1
        maxkey=0
        maxvalue=0
        for key,value in maps.items():
            if(value>maxvalue):
                maxvalue=value
                maxkey=key
        return maxkey
        

File: 169-majority-element/majority-element.java
import java.util.*;
class Solution {
    public int majorityElement(int[] nums) {
        int n=nums.length;
      
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(nums[i])==false){
                map.put(nums[i],1);

            }
            else{
                map.put(nums[i],(map.get(nums[i])+1));
            }
            if(map.get(nums[i])>n/2){
                return (nums[i]);
            }
        }
        return -1;
    }
}